docs: describe steps to install and boot Talos from the SSD on rockpi4 (#3442)
talos-systems/talos: website/content/docs/v0.10/Single Board Computers/rockpi_4.md

```bash
talosctl kubeconfig
```

> 
> ## Boot Talos from an SSD Drive
> 
> > Note: this is only tested on Rock PI 4c
> 
> Rock PI 4 has an M2 slot which supports NVMe disks. It is possible to run Talos without any SD cards right from that SSD disk.
> 
> Pre-installed SPI loader won't be able to chain Talos u-boot stored on the SSD drive because it's too outdated.
> Official docs for booting from the SSD also propose using outdated SPI flash u-boot.
> 
> To make it work it is necessary to update u-boot to a more recent version.
> Armbian u-boot for Rock PI 4c proved to be working: [https://users.armbian.com/piter75/](https://users.armbian.com/piter75/).
> 
> ### Steps
> 
> - flash any OS to the SD card (can be Armbian for example).
> - download Armbian u-boot and update SPI flash:

Reviewer: ```suggestion
## Boot Talos from an SSD Drive

> Note: this is only tested on Rock PI 4c

Rock PI 4 has an M2 slot which supports NVMe disks.
It is possible to run Talos without any SD cards right from that SSD disk.

The pre-installed SPI loader won't be able to chain Talos u-boot on the SSD drive because it's too outdated.
The official docs on booting from the SSD also propose using an outdated SPI to flash u-boot.

Instead, it is necessary to update u-boot to a more recent version for this process to work.
The Armbian u-boot build for Rock PI 4c has been proven to work: [https://users.armbian.com/piter75/](https://users.armbian.com/piter75/).

### Steps

- Flash any OS to the SD card (can be Armbian for example).
- Download Armbian u-boot and update SPI flash:
```

```bash
curl -LO https://users.armbian.com/piter75/rkspi_loader-v20.11.2-trunk-v2.img
sudo dd if=rkspi_loader-v20.11.2-trunk-v2.img of=/dev/mtdblock0 bs=4K
```

> - optionally you can also write Talos image to the SSD drive right from your Rock PI board:

Reviewer: - Optionally, you can also write Talos image to the SSD drive right from your Rock PI board:

> ## Boot Talos from an SSD Drive
> 
> > Note: this is only tested on Rock PI 4c
> 
> Rock PI 4 has an M2 slot which supports NVMe disks. It is possible to run Talos without any SD cards right from that SSD disk.
> 
> Pre-installed SPI loader won't be able to chain Talos u-boot stored on the SSD drive because it's too outdated.
> Official docs for booting from the SSD also propose using outdated SPI flash u-boot.
> 
> To make it work it is necessary to update u-boot to a more recent version.
> Armbian u-boot for Rock PI 4c proved to be working: [https://users.armbian.com/piter75/](https://users.armbian.com/piter75/).
> 
> ### Steps
> 
> - flash any OS to the SD card (can be Armbian for example).
> - download Armbian u-boot and update SPI flash:
> 
> ```bash
> curl -LO https://users.armbian.com/piter75/rkspi_loader-v20.11.2-trunk-v2.img
> sudo dd if=rkspi_loader-v20.11.2-trunk-v2.img of=/dev/mtdblock0 bs=4K
> ```
> 
> - optionally you can also write Talos image to the SSD drive right from your Rock PI board:
> 
> ```bash
> curl -LO https://github.com/talos-systems/talos/releases/latest/download/metal-rockpi_4-arm64.img.xz
> xz -d metal-rockpi_4-arm64.img.xz
> sudo dd if=metal-rockpi_4-arm64.img.xz of=/dev/nvme0n1
> ```
> 
> - remove SD card and reboot.
> 
> After that steps Talos will boot from the SSD and will enter maintenance mode. The rest of the flow is the same as for running Talos from the SD card.

Reviewer: ```suggestion
## Boot Talos from an SSD Drive

> Note: this is only tested on Rock PI 4c

Rock PI 4 has an M2 slot which supports NVMe disks.
It is possible to run Talos without any SD cards right from that SSD disk.

The pre-installed SPI loader won't be able to chain Talos u-boot on the SSD drive because it's too outdated.
The official docs on booting from the SSD also propose using an outdated SPI to flash u-boot.

Instead, it is necessary to update u-boot to a more recent version for this process to work.
The Armbian u-boot build for Rock PI 4c has been proven to work: [https://users.armbian.com/piter75/](https://users.armbian.com/piter75/).

### Steps

- Flash any OS to the SD card (can be Armbian for example).
- Download Armbian u-boot and update SPI flash:

```bash
curl -LO https://users.armbian.com/piter75/rkspi_loader-v20.11.2-trunk-v2.img
sudo dd if=rkspi_loader-v20.11.2-trunk-v2.img of=/dev/mtdblock0 bs=4K
```

- Optionally, you can also write Talos image to the SSD drive right from your Rock PI board:

```bash
curl -LO https://github.com/talos-systems/talos/releases/latest/download/metal-rockpi_4-arm64.img.xz
xz -d metal-rockpi_4-arm64.img.xz
sudo dd if=metal-rockpi_4-arm64.img.xz of=/dev/nvme0n1
```

- Remove SD card and reboot.

After these steps, Talos will boot from the SSD and enter maintenance mode.
The rest of the flow is the same as running Talos from the SD card.
```

> - remove SD card and reboot.
> 
> After that steps Talos will boot from the SSD and will enter maintenance mode. The rest of the flow is the same as for running Talos from the SD card.

Reviewer: ```suggestion
- Remove SD card and reboot.

After these steps, Talos will boot from the SSD and enter maintenance mode.
The rest of the flow is the same as running Talos from the SD card.
```
